Automattic/wp-calypso
# Store Promotions: Add variation sales to list.

This adds code to fetch variations for products and add on-sale
variations to the promotions list. Note that editing variation sales
will be covered by a PR after this one (see #20090)

This also begins using `getAllProducts` for some of the create/update
pages to prepare for de-duplicating promotion/product/coupon state.

To Test:
1. From wp-admin, add a sale price to a variation of a variable product.
2. View list of promotions, that variation sale should be visible in the list.
